x-pack/plugins/observability_solution/infra/server/routes/infra/index.ts

```diff
@@ -74,4 +83,64 @@ export const initInfraMetricsRoute = (libs: InfraBackendLibs) => {
       }
     }
   );
+
+  const validateCountBody = createRouteValidationFunction(GetInfraAssetCountRequestBodyPayloadRT);
+  const validateCountParams = createRouteValidationFunction(
+    GetInfraAssetCountRequestParamsPayloadRT
+  );
+
+  framework.registerRoute(
+    {
+      method: 'post',
```

Comment:
I wonder if we this could be a GET request.

Reply:
I need to pass the query, filters, time range, etc. so we can even expand more the body if it is a POST. I am not sure how much will those params extend in the future so I was thinking that having the consistency with the other endpoints with similar functions can help here. But I don't have a strong opinion on that. Do you think it's better to have query params instead of the request body and convert to GET?

Reply:
GET operation is preferable in this case, since nothing is being created. Infra uses `POST` for historical reasons, but most of them should be `GET` instead. The complicated part is on how to pass the filters in the query params.

Reply:
As discussed I created an issue to investigate the option to use GET instead of POST in our APIs [#189170](https://github.com/elastic/kibana/issues/189170) There are some benefits and concerns described there and it would be nice also to keep them consistent and have GET requests where possible ( I totally agree that it will be better but we still need to check if the url limitations affect us) So this change won't be part of this PR